pydantic/utils.py

```diff
@@ -155,7 +155,7 @@ def truncate(v: Union[str], *, max_len: int = 80) -> str:
     if isinstance(v, str) and len(v) > (max_len - 2):
         # -3 so quote + string + … + quote has correct length
         return (v[: (max_len - 3)] + '…').__repr__()
-    v = v.__repr__()
+    v = type(v).__repr__(v)
```

Comment:
The function's signature is
```def truncate(v: Union[str], *, max_len: int = 80) -> str:```
which is probably why I didn't think twice about changing to `v.__repr__()` from `v = repr(v)` during cythonization. Should `v` actually be annotated as `Any`?

Reply:
yes, let;s change to the try/except or add a comment or both.